cloud/ec2_fleet.go

```diff
@@ -288,6 +288,14 @@ func (m *ec2FleetManager) TerminateInstance(ctx context.Context, h *host.Host, u
 		return errors.Wrap(err, "creating client")
 	}
 
+	grip.Error(message.WrapError(disassociateIPAddressForHost(ctx, m.client, h), message.Fields{
```

Comment:
Why do we disassociate IP, terminate host, then release IP?  If there is an issue terminating, wouldn't we never release the IP?

Reply:
The job will retry as long as the host status is not "terminated", so if terminating the host errored, the entire disassociate/terminate/release process would repeat to clean up the host.

In any case, this logic isn't meant to be that bulletproof because this cleanup is prone to timing issues - calling `DisassociateAddress` _initiates_ the process of detaching the elastic IP from the network interface but the process itself happens asynchronously on EC2's side. That means this elastic IP may not be ready to release by the time `ReleaseAddress` is called. I'm going to set up a background cleanup job in DEVPROD-16716 to deal with cleaning up addresses that get left behind.

Reply:
Ah that makes sense, it might be worth a comment about this here later on when you write tests that acknowledges the timing issues but the "cleanup-elastic-ips" job covers timing issues